nexus/db-queries/src/db/datastore/snapshot.rs

```diff
@@ -113,26 +119,29 @@ impl DataStore {
                         }
                     }
 
-                    Project::insert_resource(
-                        project_id,
+                    let query = {
+                        use diesel::query_dsl::methods::FilterDsl;
+
                         diesel::insert_into(dsl::snapshot)
                             .values(snapshot)
                             .on_conflict((dsl::project_id, dsl::name))
-                            .filter_target(dsl::time_deleted.is_null())
```

Comment:
I think you wrote this as well @jmpesp -- is the replacement correct?

Reply:
I don't think so - `test_create_snapshot_record_idempotent` fails for me if I run this PR locally:

```
thread 'integration_tests::snapshots::test_create_snapshot_record_idempotent' panicked at nexus/tests/integration_tests/snapshots.rs:946:10:
called `Result::unwrap()` on an `Err` value: InternalError { internal_message: "unexpected database error: there is no unique or exclusion constraint matching the ON CONFLICT specification" }
```